### PR TITLE
Reworked javascript workflow examples to use a webserver

### PR DESCRIPTION
**Pull request by olitomlinson**
# Description

Workflows are better expressed in the context of a webserver (this is how 99% of users will host and author workflows) therefore I've included working examples of how to do this, using native express and Dapr Server.

## Issue reference

N/A

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] The quickstart code compiles correctly
* [ ] You've tested new builds of the quickstart if you changed quickstart code
* [ ] You've updated the quickstart's README if necessary
* [ ] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

### CODE REVIEW

File: workflows/javascript/sdk/order-processor/appWithDaprServer.ts
Comment (paulyuk) [inline workflows/javascript/sdk/order-processor/appWithDaprServer.ts:7]: What we do in most cases is the following with env var.

```suggestion
const DAPR_HOST = process.env.DAPR_HOST ?? "localhost" // Dapr Sidecar Host
```

**Review by paulyuk — CHANGES_REQUESTED** [review]
Small suggestion about env vars.  thanks for contribution

**Comment by paulyuk** [comment]
could you also please quick to `release-1.14` upstream branch and fix merge conflicts too @olitomlinson ?  See comment above too.

**Review by paulyuk — APPROVED** [review]
LGTM if checks pass

**Comment by paulyuk** [comment]
When someone from @dapr/approvers-workflows @dapr/maintainers-workflows has time, I cannot figure out why the initial run of the quickstart (using app.ts) is having a timeout exception, when others are not.

`dapr run -f .`

yields

```text
ℹ️  Validating config and starting app "workflowApp"
ℹ️  Started Dapr with app id "workflowApp". HTTP Port: 65495. gRPC Port: 65496
ℹ️  Writing log files to directory : /Users/paulyuk/src/oli/quickstarts/workflows/javascript/sdk/order-processor/.dapr/logs
== APP - workflowApp == 
== APP - workflowApp == > dapr-example-workflow-authoring@1.0.0 start:order-process
== APP - workflowApp == > npm run build && node dist/app.js
== APP - workflowApp == 
== APP - workflowApp == 
== APP - workflowApp == > dapr-example-workflow-authoring@1.0.0 build
== APP - workflowApp == > npx tsc --outDir ./dist/
== APP - workflowApp == 
== APP - workflowApp == 2024-07-29T22:46:56.345Z INFO [GRPCClient, GRPCClient] Opening connection to 127.0.0.1:65496
== APP - workflowApp == Successfully connected to dns:127.0.0.1:65496. Waiting for work items...
== APP - workflowApp == Workflow runtime started successfully
== APP - workflowApp == Starting new orderProcessingWorkflow instance with ID = a608a714-2569-4a25-8e66-54e29545920f
== APP - workflowApp == Orchestration scheduled with ID: a608a714-2569-4a25-8e66-54e29545920f
== APP - workflowApp == Waiting 30 seconds for instance a608a714-2569-4a25-8e66-54e29545920f to complete...
== APP - workflowApp == TimeoutError: TimeoutError
== APP - workflowApp ==     at Timeout._onTimeout (/Users/paulyuk/src/oli/quickstarts/workflows/javascript/sdk/node_modules/@microsoft/durabletask-js/client/client.js:157:68)
== APP - workflowApp ==     at listOnTimeout (node:internal/timers:569:17)
== APP - workflowApp ==     at process.processTimers (node:internal/timers:512:7)
== APP - workflowApp == Error scheduling or waiting for orchestration: TimeoutError: TimeoutError
== APP - workflowApp ==     at Timeout._onTimeout (/Users/paulyuk/src/oli/quickstarts/workflows/javascript/sdk/node_modules/@microsoft/durabletask-js/client/client.js:157:68)
== APP - workflowApp ==     at listOnTimeout (node:internal/timers:569:17)
== APP - workflowApp ==     at process.processTimers (node:internal/timers:512:7)
== APP - workflowApp == TimeoutError: TimeoutError
== APP - workflowApp ==     at Timeout._onTimeout (/Users/paulyuk/src/oli/quickstarts/workflows/javascript/sdk/node_modules/@microsoft/durabletask-js/client/client.js:157:68)
== APP - workflowApp ==     at listOnTimeout (node:internal/timers:569:17)
== APP - workflowApp ==     at process.processTimers (node:internal/timers:512:7)
The App process exited with error code: exit status 1
```

**Comment by paulyuk** [comment]
I think I fixed the grpc "Cancelled on client" and timeout issue by moving the workflowWorker.stop call outside of the start loop into a SIGTERM handler, which seems like a better idea anyway.

**Review by paulyuk — APPROVED** [review]
LGTM

**Comment by marcduiker** [comment]
@holopin-bot @olitomlinson Thanks Oli!

**Pull request merged**
